src/Hosting/DbBackup.php

```diff
@@ -0,0 +1,324 @@
+<?php
+
+/*
+ * This file is part of the Acquia Platform: Cloud Data Model.
+ *
+ * (c) Acquia, Inc.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Acquia\Platform\Cloud\Hosting;
+
+final class DbBackup implements DbBackupInterface
+{
+    /**
+     * @var int
+     */
+    private $backupId;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var int
+     */
+    private $siteId;
+
+    /**
+     * @var int
+     */
+    private $dbId;
+
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @var int
+     */
+    private $started;
+
+    /**
+     * @var int
+     */
+    private $completed;
+
+    /**
+     * @var bool
+     */
+    private $deleted;
+
+    /**
+     * @var string
+     */
+    private $checksum;
+
+    public function __construct($backupId)
+    {
+        if (!is_int($backupId)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    '%s: DbBackup ID must be an alphanumeric int (%s given)',
+                    __METHOD__,
+                    is_int($backupId) ? $backupId : gettype($backupId)
+                )
+            );
+        }
+        $this->id = $backupId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function create(array $dbBackupData)
+    {
+        $app = new static($dbBackupData['id']);
+        if (isset($dbBackupData['type'])) {
+            $app->setType($dbBackupData['type']);
+        }
+        if (isset($dbBackupData['site_id'])) {
+            $app->setSiteId($dbBackupData['site_id']);
+        }
+        if (isset($dbBackupData['db_id'])) {
+            $app->setDbId($dbBackupData['db_id']);
+        }
+        if (isset($dbBackupData['path'])) {
+            $app->setPath($dbBackupData['path']);
+        }
+        if (isset($dbBackupData['started'])) {
+            $app->setStarted($dbBackupData['started']);
+        }
+        if (isset($dbBackupData['completed'])) {
+            $app->setCompleted($dbBackupData['completed']);
+        }
+        if (isset($dbBackupData['deleted'])) {
+            $app->setDeleted($dbBackupData['deleted']);
+        }
+        if (isset($dbBackupData['checksum'])) {
+            $app->setChecksum($dbBackupData['checksum']);
+        }
```

Comment:
See https://github.com/webbj74/acquia-platform-cloud-data-model/blob/master/src/Hosting/Application.php#L87-L100 for less complex way to do this